core/src/main/java/bio/terra/pearl/core/service/participant/EnrolleeRelationService.java

```diff
@@ -45,12 +47,22 @@ public boolean isUserProxyForAnyOf(UUID participantUserId, List<UUID> enrolleeId
                 .stream().filter(enrolleeRelation -> isRelationshipValid(enrolleeRelation)).collect(Collectors.toList()).isEmpty();
     }
 
+    public Optional<Enrollee> isUserProxyForEnrollee(UUID participantUserId, String enrolleeShortcode) {
+        Enrollee targetEnrollee = enrolleeService.findOneByShortcode(enrolleeShortcode)
+                .orElseThrow(() -> new NotFoundException("Enrollee with shortcode %s was not found ".formatted(enrolleeShortcode)));
+        if(!dao.findEnrolleeRelationsByProxyParticipantUser(participantUserId, List.of(targetEnrollee.getId()))
+                .stream().filter(enrolleeRelation -> isRelationshipValid(enrolleeRelation)).collect(Collectors.toList()).isEmpty()){
```

Comment:
might be worth making a `Stream<EnrolleeRelation> validRelations(Stream<EnrolleeRelation>)` method to condense this and the logic above